cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/etl/api/batch/BatchJoinerRuntimeContext.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.etl.api.batch;
+
+import co.cask.cdap.api.data.schema.Schema;
+
+import java.util.Map;
+
+/**
+ * Runtime context for batch joiner
+ */
+public interface BatchJoinerRuntimeContext extends BatchRuntimeContext {
+
+  /**
+   * Retruns list of schemas for batch joiner
```

Comment:
typo: Retruns -> Returns. This needs some more description too, should mention that the key is the name of the input stage.  Also, what happens if that stage did not set an input schema? Is the key still in the map with a null value? Or is it not in the map at all?